### PR TITLE
WS2-1819: Added search_page and 404_page test for testing search functionality and 404 page

### PR DESCRIPTION
**Pull request by juanmitriatti**
#### Description of problem 
Write Behat tests to verify 404 page and search functionality

#### Solution
Added search_page and 404_page test for testing search functionality and 404 page

### Links

- [JIRA ticket](https://asudev.jira.com/browse/WS2-1819)

### Checklist

- [ ] Design updates match [Web Standards](https://xd.adobe.com/view/56f6cb78-9af5-4b12-b4ce-ef319f71113f-03a5/) and [Unity Design System](https://unity.web.asu.edu)
- [ ] Solution is documented on the Jira ticket
- [ ] QA steps to verify have been included on the Jira ticket
- [ ] No new PHP or JS errors
- [ ] No accessibility issues are introduced with this update
- [ ] Added/updated README.md files, if relevant

### Verified in browsers 

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Screenshots

Note: Sections that are not applicable for this issue can be removed.

### CODE REVIEW

**Comment by mlsamuelson** [comment]
[Visit Site](https://pr-316-webspark-ci.pantheonsite.io/)

Created multidev environment [pr-316](https://dashboard.pantheon.io/sites/aec2f75f-35eb-41f8-abe7-95b292415259#pr-316) for webspark-ci.

File: features/anon-user.feature [hunk 1]
```diff
@@ -13,3 +13,21 @@ Feature: Anonymous user
...
+  @api @javascript @search_page
+  Scenario: Verify Anonymous users can see and use search page
+    Given I am an anonymous user
+    When I go to the homepage
+    When I click the element ".search-button"
+    Given for "q" I enter "test"
+    Then I submit the "[name='gs']" form
+    Then the url should match "^\/\#gsc\.tab\=0\&gsc\.q=test.*?$"
+    And I should see the heading "Search"
+    Then I wait for 5 seconds
```
Comment (tbutterf) [inline features/anon-user.feature:26]: @juanmitriatti I think this ticket is meant to test the search functionality on the 404 page itself. I have added a comment on the Jira ticket itself to help clarify.

**Review by mlsamuelson — CHANGES_REQUESTED** [review]
@juanmitriatti Both of these tests are good. The ticket wasn't written clearly enough, though. @tbutterf is correct. We want to verify that the search box on the 404 page works - we actually had a regression in this functionality a few releases back so it's a good test to have.  Let's add it to this PR.

File: features/anon-user.feature (hunk 1 above)
Comment (juanmitriatti) [reply features/anon-user.feature:26]: Do you think that we have to drop the current test? Or should we keep this and add a new step for the 404 page search

Comment (mlsamuelson) [reply features/anon-user.feature:26]: @juanmitriatti The current tests are good to have. You can just add another.

**Review by mlsamuelson — APPROVED** [review]

**Pull request merged**